### PR TITLE
Demo for using CSI on Hetzner Cloud

### PR DESCRIPTION
**Pull request by pugnacity**
Simple Demo how to use CSI Plugin on Hetzner Cloud

### CODE REVIEW

File: demo/csi/hetzner/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# CSI on Hetzner Cloud
+
+This is a Terraform demo for deploying CSI volumes on Hetzner Cloud. It
+asssumes you already have a Nomad cluster running with the Docker task
```
Comment (mwild1) [inline demo/csi/hetzner/README.md:4]: Typo: asssumes -> assumes

File: demo/csi/hetzner/volume-job.nomad [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      source = "${volume_id}"
+    }
+
+    task "redis" {
+      driver = "docker"
+
+      config {
+        image = "redis:3.2"
+
+        port_map {
```
Comment (tgross) [inline demo/csi/hetzner/volume-job.nomad:16]: This is using deprecated `port_map` syntax. Let's update this to a current version of Nomad's setup for networking.

File: demo/csi/hetzner/volume-job.nomad [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        memory = 256
+
+        network {
+          mbits = 14
+          port "db" {}
+        }
+      }
+    }
+  }
+}
```
Comment (tgross) [inline demo/csi/hetzner/volume-job.nomad:37]: Let's run hclfmt on these files.

**Review by tgross — CHANGES_REQUESTED** [review]
Hi @pugnacity! We've recently merged support for `nomad volume create` but I don't think it's made it out to the Terraform provider yet (and won't be released until the upcoming Nomad 1.1.0). If the Hetnzer plugin supports `CREATE_DELETE_VOLUME` I think we'd rather have that as the workflow shown for the demo. Would you be willing to give that a go?

**Comment by pugnacity** [comment]
> Hi @pugnacity! We've recently merged support for `nomad volume create` but I don't think it's made it out to the Terraform provider yet (and won't be released until the upcoming Nomad 1.1.0). If the Hetnzer plugin supports `CREATE_DELETE_VOLUME` I think we'd rather have that as the workflow shown for the demo. Would you be willing to give that a go?

To be honest, this is my first try with the CSI plugin and I don't feel very confident to build this.
Maybe at a later stage

**Comment by cberescu** [comment]
Driver version has become outdated and so did the exmaple, the new 2.x branch has different working mechanism 

More info : https://github.com/hetznercloud/csi-driver/issues/330

**Comment by tgross** [comment]
Given the comment above about the deprecated plugin, I think I'm going to close this out unmerged. I'd also like to note that the Pack Community Registry at https://github.com/hashicorp/nomad-pack-community-registry is a great place for this kind of thing if anyone else wants to pick this up.

**Pull request closed without merging**